server/lib/nl/fulfillment/base.py

```diff
@@ -177,8 +177,155 @@ def populate_charts_for_places(state: PopulateState,
   return False
 
 
+def _add_charts_with_place_fallback(state: PopulateState, places: List[Place],
```

Comment:
nit: some comments or overall docstring might be helpful for others who might have to fiddle with this code in the future :)